### DISCUSSION TITLE
Como se sentir confortável e confiante?

### DISCUSSION THREAD

**Discussion by SpinelessCrow** [frontendbr/forum#1560] [Discussão]
Fala pessoal, gostaria de tirar umas dúvidas com vocês de como começar na carreira. Sei que já existem várias issues, li algumas e peguei um material bem bacana, mas gostaria de expor minhas dúvidas de forma mais pessoal.

Comecei um curso na Udemy, sei que não é o melhor curso mas precisava começar de algum lugar e no youtube ficava caindo na tentação de ver aulas como Introdução ao HTML do Zé e depois via Aula 1 sobre HTML do João, que era a mesma coisa, acabava me perdendo, então preferi comprar o curso, mesmo não sendo o melhor, e também nessa área o que importa é o saber(na maioria das vezes).

Primeiro: Gostaria de saber como praticar, como fazer o portfólio, como mostrar que sei o conteúdo, pode ser uma pergunta besta, mas ainda estou meio perdido. Os trabalhos do curso contam como portfólio? Mesmo sendo projetos "simples"?

Segundo: Gostaria de saber de vocês, quando vocês notaram que sabiam fazer, quando bateram no peito e disseram: Agora posso concorrer a uma vaga pois sei que posso passar. Tiveram a segurança sabe.

Obrigado pessoal, e novamente se foi repetitivo desculpem, mas estou começando e não sabia o que fazer haha.

**Comment by gabriel-hahn** [comment]
Dependendo do curso que tu pegou na Udemy, ele pode te orientar bem sobre a tecnologia que tu está estudando. Udemy tem bastante cursos bons, só tem que ter paciência para filtrar os que realmente valem a pena. Realmente as vezes o Youtube te apresenta umas dicas de vídeos que fazem a gente perder o foco hehe

Projetos de cursos "valem" também como portfólio, mas o que mais vai te ensinar e realmente dar um up no teu portfólio é teus projetos pessoais. Por exemplo, tenta montar o projeto do curso e depois tenta montar um projeto próprio utilizando as mesmas tecnologias aprendidas, assim acho uma boa maneira de realmente aprender.

Sobre a segunda pergunta, é bom distinguir entre "confiança" e o sentimento de "saber tudo". Enquanto o primeiro é algo normal até mesmo para devs experientes, o segundo pode te prejudicar bastante hehe
Confiança vai ser algo normal após estudar muito um conteúdo e realmente encarar muitos erros com uma tecnologia em si. As vezes nem é questão de conhecimento técnico, mas sim saber os caminhos de buscar as soluções dos problemas mesmo sem ter conhecimento total da linguagem/framework/lib.

**Comment by SpinelessCrow** [comment]
Cara, o curso que peguei na Udemy tem se não me engano tem 100 hrs, colei todo conteúdo aqui em baixo, sobre o portfólio, sou meio travado pq não consigo muito bem imaginar site pra criar sabe, estou com um projeto de fazer o portfólio da minha noiva com um site parallax, mas tirando isso é meio tenso. 

Sobre confiança, nós nunca vamos saber tudo haha, mas digo ter uma segurança sabe, mesmo estudando e tal...

Da uma olhada no conteúdo e me fala se o curso é bacana, acredito que seja pois um amigo meu que já foi dev falou que o curso é bem completo. Tenho planos de depois fazer um curso de JS avançado, python e talvez depois kotlin e sempre me aprimorando em todos na medida do possível

Seção 3: HTML 5

Seção 4: CSS 3

Seção 5: CSS3 Intermediário & avançado

Seção 6: HTML5 e CSS3 Recursos especiais

Seção 7: Bootstrap 4 & Design responsivo

Seção 8: JavaScript

Seção 9: JavaScript Moderno (ECMAScript 2015 - ES6) e Orientação a Objetos

Seção 10: Iniciando com servidor Apache, PHP e MySQL

Seção 11: PHP 7

Seção 12: PHP 7 e Orientação a Objetos

Seção 13: Banco de dados MySQL

Seção 14: Ajax

Seção 15: PHP com PDO e MySQL

Seção 16: Publicação de aplicações Web na Internet

Seção 17: jQuery

Seção 18: Padrão de arquitetura MVC com PHP

Seção 19: PHP 7 & MySQL - Projeto aplicado Twitter Clone

Seção 20: API com Slim Framework

Seção 21: SASS

Seção 22: Criando aplicações Mobile com IONIC

Seção 23: Wordpress

**Comment by gabriel-hahn** [comment]
100 horas é bastante extenso, talvez além do que tu realmente precise agora. Estes tipos de cursos vão te dar uma visão geral de boa parte dos ecossistemas web, envolvendo back, front e etc. Vai ser bom pra ti, mas vai te exigir bastante horas de estudos. 
Porém na minha opinião, acharia melhor tu ir entendendo as coisas em passos menores.
Ex: Se queres montar um site, procura entender (documentação ou até mesmo um curso) que fale de HTML, CSS e JavaScript somente, e tenta aplicar isso em dois ou três projetos. Após isso tu vai para um framework, para entender o real motivo de usar essas ferramentas e no que elas facilitam (não precisa entender todos, pois principalmente agora no início eles vão ser maneiras diferentes de resolver o mesmo problema pra ti). A mesma abordagem serve para o back e outros tipos de tecnologias.

Na minha opinião vale a pena ir devagar, entendendo as coisas em um ritmo lento, mas realmente entendo o motivo de cada coisa. O projeto da tua noiva já pode ser algo de portfólio, assim como um projeto simples de finanças pessoais, ou algo que faz requisição a alguma API pública, um TODO list personalizado... Não precisa ser um projeto grande, principalmente no início.
Sobre entender JS, depois Python e Kotlin, segue na mesma pegada que comentei. Vai com calma hehe dificilmente tu vai usar tudo isso de uma vez e programação é uma área impossível de dominar todas stacks existentes. Depois que tu tiver uma certa experiência e mais facilidade em entender o algoritmo e estrutura de cada linguagem, dai acho que vale mais a pena se aventurar em ferramentas diferentes.

**Comment by neriGabriel** [comment]
Opa cara bom dia, eu to trabalhando na área faz um ano e oito meses, no começo praticamente tinha as mesmas duvidas e algumas mais, algumas já "sanadas" e outras não hahaha!

Referente a portifolio, a maioria dos cursos da udemy oferece algum tipo de projeto final (simples ou complexo), é legal você subir pro seu github que já vão servir para demonstrar o seu conhecimento e que você está aprendendo, porém, não se "prenda" a isso desenvolva alguns projetos pessoais, "saia da caixa" em seus projetos porque isso vai te dar outra perspectiva e cara você está começando tente aprender o básico, o simples, exemplo: se tiver vontade de aprender css aprenda css solidifique seu conhecimento e depois vá para um framework, se for javascript a mesma coisa e assim em diante, porque apesar de tudo o framework está ali para auxiliar no desenvolvimento mas é essencial aprender o básico, lembre-se de ir devagar, com calma e tranquilo para não cair na sindrome do impostor (eu por exemplo já cai) e o mais importante acima de tudo cuide de sua saúde fisica e mental á nossa área é uma das que mais tem profissionais com depressão, _**e como algum personagem já disse em algum filme**_, você vai saber quando estiver preparado, porém lembre-se cada projeto/empresa trabalha de um jeito diferente ou seja, nunca saberemos tudo.

FAÇA NETWORK, ISSO É MUITO IMPORTANTE!!

Vou deixar uns links de alguns artigos que me ajudam/ajudaram muito:

 [A motivação desmotivacional no mundo do desenvolvimento](https://medium.com/@felipefialho/a-motiva%C3%A7%C3%A3o-desmotivacional-no-mundo-do-desenvolvimento-d0a9fcd47fd1)

[Hypes, síndrome do impostor, e o desenvolvedor junior](https://medium.com/@ricardogouveia3/hype-driven-dev-hdd-s%C3%ADndrome-do-impostor-e-o-dev-jr-c032625ccb30)

[Desenvolvedoras e desenvolvedores compartilhem conhecimento](https://www.felipefialho.com/blog/desenvolvedoras-e-desenvolvedores-do-meu-brasil-compartilhem-conhecimento/)

[15 coisas que faria se estivesse iniciando minha carreira como desenvolvedor](https://www.felipefialho.com/blog/15-coisas-que-faria-se-estivesse-iniciando-minha-carreira-como-desenvolvedor/)

[Inspirar pessoas com suas iniciativas](https://www.felipefialho.com/blog/sobre-inspirar-pessoas-com-suas-iniciativas/)

[Por que todo desenvolvedor deve ir em eventos](https://www.felipefialho.com/blog/porque-developers-devem-ir-em-eventos/)

[Por onde começar para aprender e trabalhar com programação](https://woliveiras.com.br/posts/por-onde-come%C3%A7ar-para-aprender-e-trabalhar-com-programa%C3%A7%C3%A3o/)

[Problemas ou armadilhas de sermos autodidatas](https://woliveiras.com.br/posts/problemas-ou-armadilhas-de-sermos-autodidatas/)

[Controlando a ansiedade, revisando atitudes e melhorando o auto conhecimento](https://woliveiras.com.br/posts/diário-de-bordo-controlando-a-ansiedade-revisando-atitudes-e-melhorarando-o-auto-conhecimento/)

**Comment by wilsonneto-swe** [comment]
Opa, seguindo a idéia do @neriGabriel networking é muito bom, hoje moro em sampa e a comunidade aqui é foda, vários meetups, sempre trocando notícias sobre vagas, se conhecendo e indicando.

Hoje eu já me sinto bem confiante, mas isso foi construído com projetos práticos, em empresas e pessoais mesmo, mas isso com o tempo, tenho 8 anos de experiencia com backend e atualmente uns 5 anos com back+front, e apenas 1 ano com Reac, mas ,me sinto confiante em todos (exceto react-native ainda rs).

Como comecei:
- Aprendi PHP, HTML, CSS e JS básicos
- Fiz meu blog com artigos e um CV online onde postava projetinhos para mostrar que sabia fazer rs
- Comecei a fazer alguns freelas pequenos de sites com painel de adminstração e bonitinhos (isso em 2010), pegava freela sem focar na grana mas para portifolio msm, pegava também aplicações de CRUD fáceis
- Quando tinha uns 3 sites feitos e uma aplicação de controle financeiro bem básica rodando em clientes de verdade, já me sentia bem confiante e até ganhava uma graninha já
- Comecei a correr atrás de agência/fabricas de software para entrar como estagiário ou junior em empresas, entrei como estágiario em uma empresa em Rio Preto, continuei com os freelas em PHP, mas neste ponto estava com .Net na empresa.
- Em 1 ano me dediquei ao máximo mesmo ganhando pouco ali, foi efetivado e já era um dos principais programadores ali, estudava coisa de 3h por dia, fazia freelas e o estágio. Confesso que aprendi mais com os freelas em desenvolvimento, mas aprendi mais na empresa como as coisas funcionam em empresas tradicionais.
- Dai adiante foi estudando e codificando bastante, parei com freelas mas sempre estou envolvido em algum projeto opensource ou pessoal com tecnologias diferentes das que uso na empresa para sempre evoluir...

* Eu não dominava 100% de css. js, html, php ou .net na época, sabia o suficiente para colocar sites e aplicações no ar e em produção. Boa parte do aprendizado vem na prática.

Espero que de alguma forma ajude, abs

**Comment by felipefialho** [comment]
Porra @neriGabriel, vários artigos meus citados, hahaha, fico feliz que tenham te ajudado de alguma forma.

@SpinelessCrow Dá uma olhada no [frontend-challenges](https://github.com/felipefialho/frontend-challenges), são desafios reais, se você estiver apto para fazer alguns deles está apto para trabalhar em várias empresas.

Mas vou fazer melhor e escrever um novo artigo pra te ajudar 😂

**Comment by ihenrique** [comment]
Só pra acrescentar um pouquinho, fortaleça os 3 pilares do front ( HTML , CSS ,JAVASCRIPT PURO ), sério! a galera cai de cabeça nas frameworks e esquece do básico, já entrevistei front com 9 anos de carreira que só sabia copiar e colar bootstrap e não soube resolver um problema simples com flexbox... Se você for bom nos pilares vai se sentir mais seguro durante toda a sua carreira.

**Comment by SpinelessCrow** [comment]
Opa, hoje troquei uma ideia com um cara de um trabalho antigo, ele é o responsável por entrevistar quem quer entrar na área de Front, puta responsa haha, ele me deu essa mesma dica do @ihenrique , de fortalecer os 3 pilares básicos e outra dicas que só o cara que ta a 13 anos pode dar haha.
Mas vlw gente! Estou pesquisando vários artigos também e esses que o @neriGabriel passou são bem legais e como parte são do @felipefialho muito obrigado tbm! E fico no aguardo do artigo eim rapaz!

Obrigado gente, e quanto mais puderem compartilhar melhor, não só para mim mas para outras pessoas que futuramente lerem essa Issue!

Vlw!!!

**Comment by rochabianca** [comment]
Como o pessoal tá dizendo, é de extrema importância pegar o jeito nos 3 pilares do front, mas também tem vários sites que te auxiliam pra isso, como o [Frontend Challenges do Felipe Fialho](https://github.com/felipefialho/frontend-challenges) que tem vários testes de empresa, o que é muito bom para ter certeza do que você está sabendo e o que precisa melhorar.

Eu tbm deixei algumas outras sugestões em um textinho meu, querendo ler [aqui o link](https://rochabianca.github.io/blog/como-adquirir-experiencia-em-algo-sem-trabalhar-com-isso)

**Comment by ninetails** [comment]
Confortável e confiante eu não sou até hoje rs

Como desabafo, estou participando de alguns processos seletivos e sendo rejeitado, estou amando :D (por favor, não num sentido masoquista mas sim de feedback que sei que tenho muito a crescer)

Como disse numa [outra thread](https://github.com/frontendbr/forum/issues/1550#issuecomment-536233500), faça! Não espere para achar que tá sabendo antes de tentar uma vaga. Óbvio que o básico é necessário e existem vagas Junior para vc aprender o restante, mas deixe claro que você tá aprendendo e que quer aprender muito mais. E os feedbacks de rejeição e visão de vagas de mercado vão te direcionar a próxima coisa a aprender.

<sup><sub>(E uma dica: vc paga mensalmente **0 reais** pro GitHub, o que é muita grana. Utilize-o. Faça-o ser seu backup de códigos. Seja testes, POCs, estudos, projetos simples...)</sub></sup>

**Comment by felipefialho** [comment]
> Porra @neriGabriel, vários artigos meus citados, hahaha, fico feliz que tenham te ajudado de alguma forma.
> 
> @SpinelessCrow Dá uma olhada no [frontend-challenges](https://github.com/felipefialho/frontend-challenges), são desafios reais, se você estiver apto para fazer alguns deles está apto para trabalhar em várias empresas.
> 
> Mas vou fazer melhor e escrever um novo artigo pra te ajudar 😂

Promessa é divida 💋 

[Lancei o artigo que prometi hoje](https://www.felipefialho.com/blog/como-estudar-praticar-e-conseguir-oportunidades-como-front-end-developer/)

**Comment by cassmtnr** [comment]
> Tenho planos de depois fazer um curso de JS avançado, python e talvez depois kotlin e sempre me aprimorando em todos na medida do possível.

Não tenta abraçar o mundo, foca no JavaScript no início, depois que estiver bem confiante, ataca outras tecnologias, se não você vai se frustrar em não saber o melhor que pode em nenhuma delas. O que me abriu bastante a cabeça e me tirou a falta de confiança com o JavaScript foi esse curso gratuito do Wes Bos: http://javascript30.com/.

**Comment by ROBERTOCONRADO** [comment]
Faça Projetos com tudo que aprendeu ! Faça clone de sites se possivel para fortalecer o que aprendeu de novo, só depois vá em busca de aprender algo novo.
